packages/plugin-hardhat/src/upgrade-proxy.ts

```diff
@@ -18,22 +11,30 @@ import {
   ContractAddressOrInstance,
 } from './utils';
 
+import { DeployOptions } from './deploy-proxy';
+
 export type UpgradeFunction = (
   proxy: ContractAddressOrInstance,
   ImplFactory: ContractFactory,
-  opts?: ValidationOptions,
+  opts?: DeployOptions,
```

Comment:
The reason this was using `ValidationOptions` is that the `initializer` option is not used by `upgradeProxy`.

We should do something like this:

```typescript
// common-options.ts ?
interface CommonOptions extends ValidationOptions {
   constructorArgs?: unknown[];
}

function withCommonDefaults(opts: CommonOptions): Required<CommonOptions> {
  ...
}

// deploy-proxy.ts
interface DeployOptions extends CommonOptions {
  initializer?: string | false;
}
```

etc